### PR TITLE
feat(notification): add role props

### PR DESCRIPTION
**Pull request by Thulof**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix https://github.com/ant-design/ant-design/issues/39568

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  add a new props `role` to notification component  |
| 🇨🇳 Chinese |  向通知组件添加 `role` 配置项     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by mellis481** [comment]
@Thulof Thanks for creating this PR!  When the notification has `role="status"`, you'd also need to ensure it opens in an element with `role="region"` and `aria-live="polite"` so that screen readers will (politely) announce the notification. This can be done by the following:

```
  notification.open({
    ...,
    props: {
      role: "region",
      "aria-live": "polite"
    }
  });
```
Alternatively, you could define `getContainer()` to use an HTMLElement with `role="region"` and `aria-live="polite"`.

Do you agree there isn't a better way to handle all this than these two options?

**Comment by Thulof** [comment]
> @Thulof Thanks for creating this PR! When the notification has `role="status"`, you'd also need to ensure it opens in an element with `role="region"` and `aria-live="polite"` so that screen readers will (politely) announce the notification. This can be done by the following:
> 
> ```
>   notification.open({
>     ...,
>     props: {
>       role: "region",
>       "aria-live": "polite"
>     }
>   });
> ```
> 
> Alternatively, you could define `getContainer()` to use an HTMLElement with `role="region"` and `aria-live="polite"`.
> 
> Do you agree there isn't a better way to handle all this than these two options?

@mellis481 I agree with you. I updated the interface of config

**Comment by li-jia-nan** [comment]
\rebase

File: components/notification/index.en-US.md [hunk 1]
```diff
@@ -59,6 +59,7 @@ The properties of config are as follows:
 | placement | Position of Notification, can be one of `topLeft` `topRight` `bottomLeft` `bottomRight` | string | `topRight` |
 | style | Customized inline style | [CSSProperties](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e434515761b36830c3e58a970abf5186f005adac/types/react/index.d.ts#L794) | - |
 | top | Distance from the top of the viewport, when `placement` is `topRight` or `topLeft` (unit: pixels) | number | 24 |
+| role | The semantics of notification content recognized by screen readers. The default value is alert. In this case, the screen reader will immediately interrupt other content currently being read and read the notification content instead | `alert \| status` | `alert` |
 | onClick | Specify a function that will be called when the notification is clicked | function | - |
 | onClose | Trigger when notification closed | function | - |
 | props | Props passed down | Object | An object that can contain `data-*`, `aria-*`, or `role` props, to be put on the notification `div`. This currently only allows `data-testid` instead of `data-*` in TypeScript. See https://github.com/microsoft/TypeScript/issues/28960. |
```
Comment (afc163) [inline components/notification/index.en-US.md:65]: Is it `role` inside `props`?

File: components/notification/__tests__/index.test.tsx [hunk 2]
```diff
@@ -301,4 +301,22 @@ describe('notification', () => {
...
   });
+
+  it('support role', () => {
+    act(() => {
+      notification.open({
+        message: 'Notification Title',
+        duration: 0,
+        role: 'status',
+        props: {
+          role: 'region',
```
Comment (afc163) [inline components/notification/__tests__/index.test.tsx:312]: There are two role here, which one will be applied?

**Comment by mellis481** [comment]
@Thulof After a closer look, using the existing `props` prop to add the `aria-region` and `aria-live` attributes to the container won't work. `aria-live` regions are intended to exist on the page on load (or at least before their content can be updated) and will make an announcement to screen readers when the content updates. The code that you currently have adds the `aria-live` region and the notification to the DOM at the same time. This means that the contents of the `aria-live` region never actually update and (since the notification has `role="status"` rather than `role="alert"`), the toast message is not announced at all by a screen reader.

Look at Chakra UI's [toast example](https://chakra-ui.com/docs/components/toast#usage) and test with a screen reader. The `aria-live` region exists on the page on load so that when the toast notifications (with `role="status'`) are added to the DOM _in_ that `aria-live` region, they are announced.

What I believe is needed for this feature is:

- The ability to use `role="status"` for the toast message. You've attempted to do this, but I think the ideal element for this to be added to is the one with `.ant-notification`.
- Have a single element added to the DOM which _all_ notifications can be added to. This element should:
    - have `role="region"` and `aria-live="polite"`.
    - be empty when it's created.
    - exist in the DOM before any notification can be created.

Does all this make sense?

**Comment by mellis481** [comment]
@Thulof  My apologies for the back and forth on this topic. I've unfortunately only had a few short pockets of time to think about it. 

I think all that can and should be done for this effort is to allow role=status to be set on individual notifications. You did that in your original commit satisfactorily so please revert to that commit. Ensuring the notification (with role=status) is added to a pre-existing DOM element with role=region and aria-live=polite will be up to the consumer and can be accomplished by using the getContainer prop.

**Comment by afc163** [comment]
Supported in https://github.com/ant-design/ant-design/pull/42484

**Pull request closed without merging**